clang/tools/driver/CachedDiagnostics.cpp

```diff
@@ -68,6 +76,29 @@ struct SLocEntry {
   };
 
   std::variant<FileInfo, ExpansionInfo> Data;
+
+  bool isFileInfo() const {
+    return std::holds_alternative<cached_diagnostics::SLocEntry::FileInfo>(
+        Data);
+  }
+
+  FileInfo &getAsFileInfo() {
+    assert(isFileInfo());
+    return std::get<cached_diagnostics::SLocEntry::FileInfo>(Data);
```

Comment:
I'm including the namespace for clarity when reading the code, because some of the names are same as the corresponding classes (`SLocEntry`, `FixItHint`, etc.).

Reply:
Never mind, these are right at the definitions themselves, they are unnecessary since copied over. Removed.